### DISCUSSION TITLE
About the BB2 paper

### DISCUSSION THREAD

**Discussion by tungnt55** [facebookresearch/ParlAI#4788] [Q&A]
I'm reading the paper "Beyond Goldfish Memory∗: Long-Term Open-Domain Conversation", I have two questions regarding the model and the evaluation baseline.

1.  Let's say we don't use internet search. If there is no memory existing at the current dialog turn i.e. at the beginning of the conversation or we set `knowledge_access_method` to None, then how does the model generate the response?  
2.  If I understand correctly, the baseline "Dialogue History" in Table 4 is a standard encoder-decoder that generate response from the whole dialog history. What's the correct way to replicate this model with parlai?

**Comment by mojtaba-komeili** [comment]
1. In that case we have a simple encoder-decoder model. Very similar to what we had in BB1. The context is going to be the conversation history only.
2. Don't know the answer to this one. But @jxmsML should know.

**Reply by tungnt55** [reply]
Thank you for the reply. In that case, can we get the baseline by setting `knowledge_access_method `to None? 
Or I need to train a baseline model by setting `knowledge_access_method `to None?

**Reply by mojtaba-komeili** [reply]
You don't need to train a new model.

**Reply by tungnt55** [reply]
https://github.com/facebookresearch/ParlAI/blob/4bc8f173c94359cf54e25b402d5dbca84fe6fd3d/parlai/agents/rag/modules.py#L190
I actually found the function that does the encoding of context + retrieved memory here.
Do you recommend using `input ` instead of `expanded_input ` for encoding to replicate the encoder-decoder baseline or it's better to set `knowledge_access_method` to None?
